pkg/reconciler/finalizer/controller.go

```diff
@@ -150,14 +150,10 @@ func (c *Controller) reconcileFinalizer(ctx context.Context, obj client.Object)
 		policy = metadata.DeletionPropagationPolicyOrphan
 	}
 	switch policy {
-	case metadata.DeletionPropagationPolicyForeground:
+	case metadata.DeletionPropagationPolicyForeground, metadata.DeletionPropagationPolicyOrphan:
 		if _, err := c.Finalizer.AddFinalizer(ctx, obj); err != nil {
 			return err
 		}
-	case metadata.DeletionPropagationPolicyOrphan:
-		if _, err := c.Finalizer.RemoveFinalizer(ctx, obj); err != nil {
-			return err
-		}
 	default:
 		klog.Warningf("%T %s has an invalid value for the annotation %q: %q",
```

Comment:
weird that there's a warning here. I guess we dont have any CRD field validation for it... Should we add real validation in a separate PR?

Reply:
Can you add CRD validation for annotations? Maybe using [CEL](https://kubernetes.io/docs/reference/using-api/cel/)? Either way we may not be able to add validation for backwards compatibility reasons.

Reply:
> Can you add CRD validation for annotations?

Not that I'm aware of. You get the annotations for free without specifying them in the CRD. I don't think you can add validation to unspecified fields.

We should validate them in the code, but it doesn't need to block in this PR.

Reply:
b/409592878